DotNet/Common/UnitTest/DiagnosticsTest/WindowsEventLoggerUnitTest.cs

```diff
@@ -0,0 +1,103 @@
+﻿//  --------------------------------------------------------------------------------------------------------------------
+//  <copyright file="WindowsEventLoggerUnitTest.cs" company="CloudLibrary">
+//    Copyright (c) CloudLibrary 2016.  All rights reserved.
+//  </copyright>
+//  --------------------------------------------------------------------------------------------------------------------
+
+namespace CloudLibrary.Common.UnitTest.DiagnosticsTest
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Diagnostics.CodeAnalysis;
+
+    using Common.Diagnostics;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Shared.CommonUnitTestHelper;
+    using Shared.UnitTestHelper;
+
+    /// <summary>
+    /// Windows Event Logger unit test
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    [TestClass]
+    public class WindowsEventLoggerUnitTest
+    {
+        /// <summary>
+        /// Test Windows Event Logger
+        /// </summary>
+        [TestMethod]
+        public void WindowsEventLogger_UnitTest()
+        {
+            var testCount = 3;
+            var random = new Random();
+
+            var logger = new WindowsEventLogger();
+            var stackTrace = new StackTrace(true);
+            var runtimeInformation = new Dictionary<string, object>();
+
+            // Run through the test
+            for (var testId = 0; testId < testCount; testId++)
+            {
+                var eventId = Math.Abs(random.Next()) % 256;
+                var allEventLevels = EnumHelper.GetAllPossibleValues<EventLevel>();
+
+                // Test all event levels
+                foreach (var eventLevel in allEventLevels)
+                {
+                    var eventSource = TestHelper.CreateUniqueName("SourceName{0}");
```

Comment:
No sure whether this will work because the event source is a random string and windows event log system may don't have this event source registered. Please double check.